### PR TITLE
Update log level for successful retry

### PR DESCRIPTION
**Pull request by shivamkm07**
# Description

In case of retries, pubsub.azure.eventhub returns error even after successful event processing: https://github.com/dapr/components-contrib/blob/2b5650c2faa41f0a8d82862583497103776be49f/pubsub/azure/eventhubs/eventhubs.go#L605

It should be a Warning and not an Error

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by mukundansundar — DISMISSED** [review]

**Comment by shivamkm07** [comment]
> If I am seeing the original error I also want to see the recovery with the same log level. otherwise the original error should have been warn level also since we were able to recover. Maybe only a complete failure should use log level error.

So, should both be warn levels? Some of the other components are having different levels, like https://github.com/dapr/components-contrib/blob/2b5650c2faa41f0a8d82862583497103776be49f/bindings/alicloud/rocketmq/rocketmq.go#L291
https://github.com/dapr/components-contrib/blob/2b5650c2faa41f0a8d82862583497103776be49f/internal/component/kafka/consumer.go#L43

**Review by artursouza — DISMISSED** [review]

**Comment by artursouza** [comment]
I think we should leave the retriable error as log error because when the message is given up, we would not have any log err.

**Comment by berndverst** [comment]
Can we make the log level consistent please? @artursouza - should retriable errors log as error? I think warning or debug seems better? Thoughts?

**Comment by berndverst** [comment]
> I think we should leave the retriable error as log error because when the message is given up, we would not have any log err.

I see. Makes sense. I think that's not ideal when things are retried and recover. We should change that somehow in our resiliency framework.

**Comment by mukundansundar** [comment]
Can we make it consistent across components? Possibly a helper function to clearly wrap/log events in a consistent fashion similar to retry notify recover wrapper?

**Comment by berndverst** [comment]
> Can we make it consistent across components? Possibly a helper function to clearly wrap/log events in a consistent fashion similar to retry notify recover wrapper? 

That's a separate question which does not belong on this issue @mukundansundar

**Review by berndverst — CHANGES_REQUESTED** [review]
Please make both a warn here.

Then below in line 608 log the actual error!

File: pubsub/azure/eventhubs/eventhubs.go [hunk 1]
```diff
@@ -602,7 +602,7 @@ func (aeh *AzureEventHubs) Subscribe(subscribeCtx context.Context, req pubsub.Su
 			}, b, func(_ error, _ time.Duration) {
 				aeh.logger.Errorf("Error processing EventHubs event: %s/%s. Retrying...", req.Topic, e.ID)
```
Comment (berndverst) [inline pubsub/azure/eventhubs/eventhubs.go:603]: This should also be WarnF

File: pubsub/azure/eventhubs/eventhubs.go [hunk 2]
```diff
@@ -602,7 +602,7 @@ func (aeh *AzureEventHubs) Subscribe(subscribeCtx context.Context, req pubsub.Su
...
 			}, func() {
-				aeh.logger.Errorf("Successfully processed EventHubs event after it previously failed: %s/%s", req.Topic, e.ID)
+				aeh.logger.Warnf("Successfully processed EventHubs event after it previously failed: %s/%s", req.Topic, e.ID)
 			})
 		})
 	if err != nil {
```
Comment (berndverst) [inline pubsub/azure/eventhubs/eventhubs.go:608]: This is where we want to log the only Errorf as this indicates that the retries have been exhausted.

**Review by berndverst — CHANGES_REQUESTED** [review]
See comments.

File: pubsub/azure/eventhubs/eventhubs.go (hunk 2 above)
Comment (shivamkm07) [reply pubsub/azure/eventhubs/eventhubs.go:612]: Done

File: pubsub/azure/eventhubs/eventhubs.go (hunk 1 above)
Comment (shivamkm07) [reply pubsub/azure/eventhubs/eventhubs.go]: Done

**Comment by shivamkm07** [comment]
> > I think we should leave the retriable error as log error because when the message is given up, we would not have any log err.
> 
> I checked with @halspang and this is actually incorrect. We can log something separately after the retries are exhausted. So we should update this to be WarnF in the retry block, and then below we should add a new log statement for the ErrorF. IMO that is the right way to handle this.

Done

**Review by ItalyPaleAle — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

**Pull request merged**